.github/workflows/golangci-lint.yml

```
@@ -0,0 +1,50 @@
+name: Lint
+on:
+  push:
+    branches:
+      - master
+      - main
+  pull_request:
+permissions:
+  contents: read
+  # Optional: allow read access to pull request. Use with `only-new-issues` option.
+  # pull-requests: read
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event_name == 'pull_request' && github.head_ref || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  golangci:
```

I’d probably make the job key consistent with the `name`
```suggestion
  lint:
```